### PR TITLE
ENH: Series.explode to support pyarrow-backed list types

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/series/methods/test_explode.py [hunk 1]
```diff
@@ -141,3 +143,25 @@ def test_explode_scalars_can_ignore_index():
     result = s.explode(ignore_index=True)
     expected = pd.Series([1, 2, 3])
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.skipif(pa_version_under7p0, reason="minimum pyarrow not installed")
```
Comment (mroeschke) [inline pandas/tests/series/methods/test_explode.py:148]: Nit: Could you just use `pa = pytest.importorskip("pyarrow")` below?

File: pandas/tests/series/methods/test_explode.py [hunk 2]
```diff
@@ -141,3 +143,25 @@ def test_explode_scalars_can_ignore_index():
...
+def test_explode_pyarrow_list_type():
```
Comment (mroeschke) [inline pandas/tests/series/methods/test_explode.py:149]: Could you parameterize on `ignore_index` as well?

Comment (lukemanley) [reply pandas/tests/series/methods/test_explode.py]: updated

**Review by mroeschke — COMMENTED** [review]
Looks good just a merge conflict

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Pull request merged**